hubbard/hamiltonian.py

```diff
@@ -259,13 +264,13 @@ def update_hamiltonian(self):
 
         q0 = self.geometry.atoms.q0
         E = self.e0.copy()
-        E += self.U * (self.n[[1, 0], :] - q0)
+        E += self.U * (self.n[::-1, :] - q0)
```

Comment:
This may be problematic if the spin dimensionality is higher than two and/or if the basis is non-orthogonal (where the last element is the overlap).